Lib/test/test_typing.py

```diff
@@ -2778,6 +2778,15 @@ class HasForeignBaseClass(mod_generics_cache.A):
 
 gth = get_type_hints
 
+class ForRefExample():
+    @ann_module.dec
+    def func(a: 'ForRefExample'):
```

Comment:
Minor, but maybe just name this argument `self` for cleanliness?

```suggestion
    def func(self: 'ForRefExample'):
```